chimney/src/main/scala/io/scalaland/chimney/TransformerFSupport.scala

```diff
@@ -0,0 +1,97 @@
+package io.scalaland.chimney
+
+import scala.collection.compat._
+import scala.language.higherKinds
+
+trait TransformerFSupport[F[_]] {
+
+  def pure[A](value: A): F[A]
+
+  def combine[A, B](fa: F[A], fb: => F[B]): F[(A, B)]
+
+  def map[A, B](fa: F[A], f: A => B): F[B]
+
+  def traverse[M[X] <: Iterable[X], A, B](mfa: M[A], f: A => F[B])(
+      implicit fac: Factory[B, M[B]]
+  ): F[M[B]]
+}
+
+object TransformerFSupport {
+
+  implicit def TransformerFOptionSupport: TransformerFSupport[Option] =
+    new TransformerFSupport[Option] {
+
+      def pure[A](value: A): Option[A] = Option(value)
+
+      def combine[A, B](fa: Option[A], fb: => Option[B]): Option[(A, B)] = {
+        fa.flatMap(a => fb.map(b => (a, b)))
```

Comment:
Unfortunately, it doesn't work with Scala 2.11/2.12:
```
[error] TransformerFSupport.scala:27:15: type mismatch;
[error]  found   : Iterable[(A, B)]
[error]  required: Option[(A, B)]
[error]         fa.zip(fb)
[error]               ^
[info] Iterable[(A, B)] <: Option[(A, B)]?
[info] false
```

It seems that `.zip` on `Option` was added in 2.13 during collections rework. In 2.11/2.12 it was available through optionToIterable implicit conversion, but resulting type is Iterable - hence the type error.

It could be fixed by using `.headOption` that compiles fine on all 2.11-2.13:
```scala
fa.zip(fb).headOption
```

But I'm not sure whether it's good idea as there is additional iterator materialization under the hood.

Alternative possible implementation which is more like `.zip` from 2.13:

```scala
def combine[A, B](fa: Option[A], fb: => Option[B]): Option[(A, B)] = {
  lazy val fbv = fb
  if (fa.isEmpty || fbv.isEmpty) None else Some((fa.get, fbv.get))
}
```

Reply:
> it doesn't work with Scala 2.11/2.12

I didn't know that.

```scala
fa.flatMap(a => fb.map(b => (a, b)))
```
much better I think 😄